Runtime/Scripts/LyumaAv3Emulator.cs

```diff
@@ -187,102 +185,126 @@ private void Start()
 			Camera.onPreCull += PreCull;
 			Camera.onPostRender += PostRender;
 			emulatorInstance = this;
-			RunPreprocessors();
 			if (WorkaroundPlayModeScriptCompile) {
 				LyumaAv3Runtime.ApplyOnEnableWorkaroundDelegate();
 			}
-
-			InitNextFrame = true;
+			AddMarkersToDisabled();
+			InitializeInScene();
 		}
 
-		private void RunPreprocessors()
+		[DefaultExecutionOrder(-9999)]
+		private class LyumaAv3Marker : MonoBehaviour
 		{
-			List<GameObject> avatars = AvatarList;
-			foreach (var avadesc in avatars)
+			private void Start()
 			{
-				bool alreadyHadComponent = avadesc.gameObject.GetComponent<LyumaAv3Runtime>() != null;
-				if (RunPreprocessAvatarHook && !alreadyHadComponent)
+				if (GetComponent<VRCAvatarDescriptor>() != null)
 				{
-					GameObject origClone = GameObject.Instantiate(avadesc.gameObject);
-					origClone.name = avadesc.gameObject.name;
-					avadesc.gameObject.name = origClone.name + "(Clone)";
-					LyumaAv3Runtime.InvokeOnPreProcessAvatar(avadesc.gameObject);
-					avadesc.gameObject.name = origClone.name;
-					GameObject.DestroyImmediate(origClone);
+					Debug.Log("Setting up Emulator late on: " + gameObject.name);
+					emulatorInstance.RunPreprocessors(gameObject);
+					emulatorInstance.Initialize(gameObject);
 				}
 			}
 		}
 		
-		private void Initialize()
+		private void AddMarkersToDisabled()
+		{
+			Resources.FindObjectsOfTypeAll<VRCAvatarDescriptor>()
```

Comment:
according to docs, this includes prefabs so I think we should exclude [Persistent](https://docs.unity3d.com/2020.3/Documentation/ScriptReference/EditorUtility.IsPersistent.html) objects